perftests/Xcode/PerfTests/BuildSystemPerfTests.mm

```diff
@@ -109,6 +114,10 @@ - (void)testSupprocessSpawnWorkingDirectory {
             spawnProcess(delegate, nullptr, pgrp, handle, cmd, environment, attr, std::move(releaseFn), std::move(completionFn));
         }
     }];
+
+    // Reset (remove) per-thread working directory in case it was set by the
+    // above process spawning.
+    __pthread_fchdir(-1);
```

Comment:
Should this go in the process spawning code rather than here?

Reply:
The process code currently is optimized for common case performance when used under the execution queue.  If we expand our use of the subprocess code, we should revisit that behavior. For our current use of it, I think it is fine to leave it as is.